### ISSUE TITLE
withFeatureCheck is not a function (v2.0.0-alpha.2)

### ISSUE THREAD

**Issue by lnhrdt** [faisalman/ua-parser-js#656]
I learned about the new `withClientHints` and `withFeatureCheck` helpers in your alpha releases notes & docs:
- https://github.com/faisalman/ua-parser-js/releases/tag/2.0.0-alpha.2
- https://faisalman.github.io/ua-parser-js-docs/v2/api/ua-parser-js/idata/with-client-hints.html
- https://faisalman.github.io/ua-parser-js-docs/v2/api/ua-parser-js/idata/with-feature-check.html

The `withClientHints` seems to work:

```typescript
UAParser(window.navigator.userAgent)
  // @ts-ignore
  .withClientHints()
  .then((result: UAParser.IResult) => {
    console.log('UAParser result:', result)
  })
````

But the `withFeatureCheck` method is not found:

```typescript
UAParser(window.navigator.userAgent)
  // @ts-ignore
  .withFeatureCheck()
  .then((result: UAParser.IResult) => {
    console.log('UAParser result:', result)
  })
````

error message:
> Uncaught (in promise) TypeError: UAParser(...).withFeatureCheck is not a function

My environment (browser):
- `ua-parser-js`: `2.0.0-alpha.2`
- `@types/ua-parser-js`: `0.7.36`
- Google Chrome: Version 113.0.5672.126 (Official Build) (arm64)

**Comment by rajeakshay** [comment]
I was also interested in trying out `withFeatureCheck` but `2.0.0-alpha.2` does not have it - https://github.com/faisalman/ua-parser-js/blob/2.0.0-alpha.2/src/ua-parser.mjs . It is in a later commit  - https://github.com/faisalman/ua-parser-js/commit/801c2409b3e0fb83f67e95ebeca2723adb4976c9 .

@faisalman Any plans to create another alpha release so early adopters like us can try it out?

**Comment by lnhrdt** [comment]
@rajeakshay thank you for sharing your discovery! I was spinning my wheels trying to figure out what I was doing wrong.

@faisalman I agree another alpha release would be great. Something else to consider, it might be worth aligning your public docs with the released versions. Or at least changing to the docs to indicate they don't represent a released version. At the moment, the docs I linked above say they're for `2.0.0-alpha` and that's where I got misled today.

**Comment by rdumouchelle** [comment]
Seconding above ^^ another release to address this would be appreciated.

**Comment by aPaladinesEC** [comment]
Hi, the same here, unavailable to test withFeatureCheck using the pre-release 2.0.0-alpha.2

**Comment by faisalman** [comment]
`withFeatureCheck` method has now been included in the recently published `v2.0.0-alpha.3`

**Comment by th0rgall** [comment]
FYI: I'm still getting the same error with the recently released `2.0.0-beta.1`, and with the previously released `-alpha.3`.

In contrast to the code in the opening post here, the docs also seem to suggest a synchronous, and not asynchronous use of this API?

```
// suppose this code runs on iPad
const withoutFeatureCheck = UAParser();
const withFeatureCheck = UAParser().withFeatureCheck();

console.log(withoutFeatureCheck.device); // { vendor : "Apple", model : "Macintosh", type : undefined }
console.log(withFeatureCheck.device);    // { vendor : "Apple", model : "iPad", type : "tablet" }
```

Is that working for someone?
